test/e2e_node/remote/node_e2e.go

```diff
@@ -114,6 +124,7 @@ func osSpecificActions(args, host, workspace string) (string, error) {
 		return args, setKubeletSELinuxLabels(host, workspace)
 	case strings.Contains(output, "gci"), strings.Contains(output, "cos"):
 		args = prependMemcgNotificationFlag(args)
+		args = prependGCPCredentialProviderFlag(args, workspace)
```

Comment:
same, are we prepending the flags always?

Reply:
There are two horizons - short term and longer term. In short term this job will likely run in separate test definition so simply making sure that it only being pre-pended on COS is OK.

Longer term we would like this job to be a NodeConformance job as it test non-optional feature of kubelet. With this, it will likely need to run on Ubuntu and COS. So it will be best to prepend this when the cloud provider set to GCE.

If we are in rush with this to get the KEP moving, I'd be fine keeping it as is. But I'd like to have this added to the GA requirement - to include this test into NodeConformance lane.

Reply:
> If we are in rush with this to get the KEP moving, I'd be fine keeping it as is. But I'd like to have this added to the GA requirement - to include this test into NodeConformance lane.

sgtm!

Reply:
I guess there is no harm in prepending to ubuntu as well.

Reply:
And also can we add a comment something `prepending flag here in test framework is necessary as we don't know the location of credential provider binary and config before node vm startup so adding the flags via job config is not feasible`